Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -273,7 +273,7 @@ int ParticleBoundaryBuffer::getNumParticlesInContainer(
     else return 0;
 }
 
-WarpXParticleContainer::ContainerLike<amrex::PinnedArenaAllocator> &
+PinnedMemoryParticleContainer &
```

Comment:
The corresponding `.H` file indicates that this function should return a `PinnedMemoryParticleContainer`.
Now that `PinnedMemoryParticleContainer` is a type of `NamedComponentParticleContainer`, it is not equivalent anymore to use `WarpXParticleContainer::ContainerLike<amrex::PinnedArenaAllocator>` instead of `PinnedMemoryParticleContainer`.

Reply:
Should we move `ContainerLike` from `WarpXParticleContainer` up to `NamedComponentParticleContainer` now?

Reply:
`ContainerLike` is actually a type/function that is defined in `amrex::ParticleContainer`, I think, and then `WarpXParticleContainer` inherits it.

Reply:
Oh, I added it there, hah. Overlooked, sorry :)